htmresearch/frameworks/layers/l2_l4_network_creation.py

```diff
@@ -362,70 +364,132 @@ def createMultipleL4L2Columns(network, networkConfig):
-            of columns will be connected to each other.  Useful specific values
-            are 1 and 1.5, which typically create grids without and with
-            diagonal connections, respectively.
-        "L4Params": {
-          <constructor parameters for ExtendedTMRegion>
-        },
-        "L2Params": {
-          <constructor parameters for ColumnPoolerRegion>
-        },
-        "lateralSPParams": {
-          <constructor parameters for optional SPRegion>
-        },
-        "feedForwardSPParams": {
-          <constructor parameters for optional SPRegion>
-        }
-      }
-    """
-
-    # Create each column
-    numCorticalColumns = networkConfig["numCorticalColumns"]
-    for i in xrange(numCorticalColumns):
-      networkConfigCopy = copy.deepcopy(networkConfig)
-      layerConfig = networkConfigCopy["L2Params"]
-      layerConfig["seed"] = layerConfig.get("seed", 42) + i
-
-      layerConfig["numOtherCorticalColumns"] = numCorticalColumns - 1
-
-      suffix = "_" + str(i)
-      network = createL4L2Column(network, networkConfigCopy, suffix)
-
-    # Now connect the L2 columns laterally
-    for i, src_pos in enumerate(networkConfig["columnPositions"]):
-      suffixSrc = "_" + str(i)
-      for j, dest_pos in enumerate(networkConfig["columnPositions"]):
-        if i != j and numpy.linalg.norm(numpy.asarray(src_pos) -
-             numpy.asarray(dest_pos)) <= networkConfig["maxConnectionDistance"]:
-          suffixDest = "_" + str(j)
-          network.link(
-            "L2Column" + suffixSrc, "L2Column" + suffixDest, "UniformLink", "",
-            srcOutput="feedForwardOutput", destInput="lateralInput",
-            propagationDelay=1)
-
-    enableProfiling(network)
-
-    return network
+  """
+  Create a network consisting of multiple columns.  Each column contains one
+  L4 and one L2, is identical in structure to the network created by
+  createL4L2Column. In addition the L2 columns are connected to each
+  other through their lateral inputs, based on the topological information
+  provided.
+
+  Region names have a column number appended as in externalInput_0,
+  externalInput_1, etc.
+
+  networkConfig must be of the following format (see createL4L2Column for
+  further documentation):
+
+    {
+    "networkType": "MultipleL4L2Columns",
+    "numCorticalColumns": 3,
+    "externalInputSize": 1024,
+    "sensorInputSize": 1024,
+    "columnPositions": a list of 2D coordinates, one for each column.
+      Used to calculate the connections between columns. By convention,
+      coordinates are integers.
+    "maxConnectionDistance": should be a value >= 1.  Determines how distant
+      of columns will be connected to each other.  Useful specific values
+      are 1 and 1.5, which typically create grids without and with
+      diagonal connections, respectively.
+    "longDistanceConnections": If this is a value in [0,1), it is taken
```

Comment:
It's not exceptionally important now, but it could be useful for testing cases in which we want a more predictable network structure, or if we start producing very large networks and want to keep the degree to which columns cross-link fixed.  I'll cut it out now, and if we need it later we can reimplement it then.